test/integration/test_config_defaults.py

```diff
@@ -1,22 +1,191 @@
-        return self._app.config.appschema[key]['default']
+from galaxy.util import listify
 
-    def test_default_database_engine_option_pool_recycle(self):
-        expect = self.get_default('database_engine_option_pool_recycle')
-        assert expect == self._app.config.database_engine_option_pool_recycle
 
-    def test_default_database_engine_option_server_side_cursors(self):
-        expect = self.get_default('database_engine_option_server_side_cursors')
-        assert expect == self._app.config.database_engine_option_server_side_cursors
+"""
+This tests: (1) automatic creation of configuration properties; and
+(2) assignment of default values that are specified in the schema and, in
+some cases, are also processed at load time (paths resolved, csv strings
+converted to lists, etc).
 
-    def test_default_database_query_profiling_proxy(self):
-        expect = self.get_default('database_query_profiling_proxy')
-        assert expect == self._app.config.database_query_profiling_proxy
+This module will test ALL schema properties, unless they are listed in the
+global DO_NOT_TEST. Whenever a property's default value is changed (edits to
+schema or configuration loading procedures), this test code must be modified to
+reflect that change.
+
+Test assumptions for a default configuration:
+- If a default is set and not modified at load time, expect schema default.
+- If a default is not set, expect null.
+- If a default is set and modified at load time, the test should reflect that
+  (if a default is specified in the schema, it is expected that it will be used
+  in some form at load time; otherwise it should not be listed as a default).
+
+Configuration options NOT tested:
+- config_dir (value overridden for testing)
+- data_dir (value overridden for testing)
+- new_file_path (value overridden for testing)
+- logging (mapping loaded in config/; TODO)
+- dependency_resolution (nested properties; TODO)
+- job_config (no obvious testable defaults)
+"""
+
+
+OptionData = namedtuple('OptionData', 'key, expected, loaded')
+
+
+# Most of these (except root_dir) will go away once path_resolves_to is set in the schema
+RESOLVE = {
+    'sanitize_whitelist_file': 'root_dir',
+    'tool_data_path': 'root_dir',
+    'involucro_path': 'root_dir',
+    'tool_path': 'root_dir',
+    'integrated_tool_panel_config': 'config_dir',
+    'shed_tool_data_path': 'tool_data_path',
+    'builds_file_path': 'tool_data_path',
+    'len_file_path': 'tool_data_path',
+}
+
+CUSTOM = {
+    'password_expiration_period': timedelta,
+    'toolbox_filter_base_modules': listify,
+    'mulled_channels': listify,
+    'user_library_import_symlink_whitelist': listify,
+    'tool_filters': listify,
+    'tool_label_filters': listify,
+    'tool_section_filters': listify,
+    'persistent_communication_rooms': listify,
+}
+
+# TODO: split into (1) do not test; and (2) todo: fix and test
+DO_NOT_TEST = [
+    'config_dir',  # value overridden for testing
```

Comment:
So now that we have a nice getter/setter system we could probably also record when something is not the default value.
Not for this PR of course.

Reply:
Yes, I'm sure we can! And I'm certainly not done with this -- but yes, that would be for a subsequent PR. (as well as cleaning up/normalizing the getter setter system)